mon-pix/app/components/challenge-item-generic.js

```diff
@@ -123,5 +133,29 @@ export default class ChallengeItemGeneric extends Component {
   setUserConfirmation() {
     this._start();
     this.hasUserConfirmedWarning = true;
+    this._addQuitTimedChallengeListeners();
+  }
+
+  _removeEventListeners() {
+    if (this._pageBeforeUnloadListener) {
+      window.removeEventListener('beforeunload', this._pageBeforeUnloadListener);
+    }
+    if (this._pageUnloadListener) {
+      window.removeEventListener('unload', this._pageUnloadListener);
+    }
+  }
+
+  _addQuitTimedChallengeListeners() {
+    this._pageBeforeUnloadListener = (event) => {
+      event.preventDefault();
+    };
+    this._pageUnloadListener = (event) => {
+      event.preventDefault();
+      if (!this._hasAbandOrValidateAnswer) {
+        this.skipChallenge();
+      }
+    };
+    window.addEventListener('beforeunload', this._pageBeforeUnloadListener);
+    window.addEventListener('unload', this._pageUnloadListener);
```

Comment:
```suggestion
  _addQuitTimedChallengeListeners() {
    this._pageBeforeUnloadListener = (event) => {
      event.preventDefault();
    };
    this._pageUnloadListener = (event) => {
      event.preventDefault();
      if (!this._hasAbandOrValidateAnswer) {
        this.skipChallenge();
      }
    };
    window.addEventListener('beforeunload', this._pageBeforeUnloadListener);
    window.addEventListener('unload', this._pageUnloadListener);
  }

  _removeEventListeners() {
    if (this._pageBeforeUnloadListener) {
      window.removeEventListener('beforeunload', this._pageBeforeUnloadListener);
    }
    if (this._pageUnloadListener) {
      window.removeEventListener('unload', this._pageUnloadListener);
    }
  }
```

**Suggestion:** Inverser l'ordre des appels afin de mieux retranscrire l'ordre des évènements.